test/libsolidity/Metadata.cpp

```diff
@@ -225,6 +225,45 @@ BOOST_AUTO_TEST_CASE(metadata_stamp_experimental)
 		}
 }
 
+BOOST_AUTO_TEST_CASE(metadata_eof_experimental)
+{
+	// Check that setting an EOF version results in the experimental flag being set.
+	char const* sourceCode = R"(
+		pragma solidity >=0.0;
+		contract test {
+			function g(function(uint) external returns (uint) x) public {}
+		}
+	)";
+	for (auto metadataFormat: std::set<CompilerStack::MetadataFormat>{
+		CompilerStack::MetadataFormat::NoMetadata,
+		CompilerStack::MetadataFormat::WithReleaseVersionTag,
+		CompilerStack::MetadataFormat::WithPrereleaseVersionTag
+	})
+	{
+		CompilerStack compilerStack;
+		compilerStack.setMetadataFormat(metadataFormat);
+		compilerStack.setSources({{"", sourceCode}});
+		compilerStack.setEVMVersion(langutil::EVMVersion::prague());
+		compilerStack.setViaIR(true);
+		compilerStack.setEOFVersion(1);
+		compilerStack.setOptimiserSettings(true);
```

Comment:
To get more coverage, we should use values from the command-line rather than hard-coded ones:

```suggestion
		compilerStack.setEVMVersion(CommonOptions::get().evmVersion());
		compilerStack.setViaIR(true);
		compilerStack.setEOFVersion(CommonOptions::get().eofVersion());
		compilerStack.setOptimiserSettings(CommonOptions::get().optimize);
```

Then for non-EOF just expect experimental to be not set.

Note that this is the way most other tests here already work here.